src/panels/lovelace/cards/hui-logbook-card.ts

```diff
@@ -0,0 +1,312 @@
+    if (
+      changedProps.has("_config") ||
+      changedProps.has("_persons") ||
+      changedProps.has("_logbookEntries")
+    ) {
+      return true;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+
+    if (
+      !this._configEntities ||
+      !oldHass ||
+      oldHass.themes !== this.hass!.themes ||
+      oldHass.language !== this.hass!.language
+    ) {
+      return true;
+    }
+
+    for (const entity of this._configEntities) {
+      if (oldHass.states[entity.entity] !== this.hass!.states[entity.entity]) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  protected firstUpdated(): void {
+    this._fetchPersonNames();
+  }
+
+  protected updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const configChanged = changedProperties.has("_config");
+    const hassChanged = changedProperties.has("hass");
+    const oldHass = changedProperties.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProperties.get("_config") as LogbookCardConfig;
+
+    if (
+      (hassChanged && oldHass?.themes !== this.hass.themes) ||
+      (configChanged && oldConfig?.theme !== this._config.theme)
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config.theme);
+    }
+
+    if (
+      configChanged &&
+      (oldConfig?.entities !== this._config.entities ||
+        oldConfig?.hours_to_show !== this._config!.hours_to_show)
+    ) {
+      this._logbookEntries = undefined;
+      this._lastLogbookDate = undefined;
+
+      if (!this._configEntities) {
+        return;
+      }
+
+      this._throttleGetLogbookEntries();
+      return;
+    }
+
+    if (
+      oldHass &&
+      this._configEntities!.some(
+        (entity) =>
+          oldHass.states[entity.entity] !== this.hass!.states[entity.entity]
+      )
+    ) {
+      // wait for commit of data (we only account for the default setting of 1 sec)
+      setTimeout(this._throttleGetLogbookEntries, 1000);
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    if (!isComponentLoaded(this.hass, "logbook")) {
+      return html`
+        <hui-warning>
+          ${this.hass.localize(
+            "ui.components.logbook.component_not_loaded"
+          )}</hui-warning
+        >
+      `;
+    }
+
+    return html`
+      <ha-card
+        .header=${this._config!.title}
+        class=${classMap({ "no-header": !this._config!.title })}
+      >
+        <div class="content">
+          ${!this._logbookEntries
+            ? html`
+                <ha-circular-progress
+                  active
+                  alt=${this.hass.localize("ui.common.loading")}
+                ></ha-circular-progress>
+              `
+            : this._logbookEntries.length
+            ? html`
+                <ha-logbook
```

Comment:
Should we virtualize this list?

Reply:
idk how to do that :)

Reply:
Add the `virtualize` attribute ;-) https://github.com/home-assistant/frontend/blob/dev/src/panels/logbook/ha-logbook.ts#L92-L93